### DISCUSSION TITLE
Protein folding siimulation in TorchMD

### DISCUSSION THREAD

**Discussion by Lcressot** [torchmd/torchmd#50] [Q&A]
Hello,
I am working on a project involving molecular simulation and zero proof knowledge. We need a TorchMD model taking in input a protein folding and computing its energy state (which be be lower when the folding is good). Any one could tell how I could do this exactly ?
Thanks so much !

**Comment by giadefa** [comment]
I don't understand what you are trying to do

**Reply by Lcressot** [reply]
Hey, thanks for replying.
So, given a file with the positions of atoms of a folded protein, I would like to compute the energy state of this folding. If the protein is well folded (like they do in real life), then this energy should be minimal. (Google alpha fold was trained to minimize this energy).

And I need to make this computation with a TorchMD model (there is a reason why I need TorchMD and no other tools, but that's not of importance for you).

Thanks a lot !

**Comment by stefdoerr** [comment]
https://github.com/torchmd/torchmd/blob/master/examples/tutorial.ipynb
There is a tutorial which shows how to calculate the energy

**Reply by LokartBC** [reply]
Hey, thanks a lot, that looks very interesting

**Comment by julien-blanchon** [comment]
What's the point of using zkproof with molecular simulation ?

**Reply by LokartBC** [reply]
Hi, the idea is to be able to prove one detains some information like a protein folding or a given drug, without revealing this information. It can be used to protect knowledge and discuss with investors etc
